### PR TITLE
Add generation test

### PR DESCRIPTION
**Pull request by dtan4**
## WHY
We already have some unit tests to check the below:

- Files are generated in appropriate location
- File body corresponds to the expected one
- Method returns the expected response

However, we also have to check the behavior of generator itself. Additionally, API server in `_example` should be maintained as the latest artifacts.

## WHAT
Add intergation test and run at CI. This test does:

0. Regenerate all files of API server in `_example` 
0. Check whether there is no diff between before and after
0. Try building API server and check whether build succeeds

⚠️ This pull request depends on changes in #53 . After merging #53 I'll update files of API server in `_example` and remove `[WIP]` flag.

### CODE REVIEW

**Comment by dtan4** [comment]
@awakia Please review this 🙇

File: _example/router/router.go [hunk 1]
```diff
@@ -10,12 +10,24 @@ func Initialize(r *gin.Engine) {
 	api := r.Group("api")
 	{
 
+		api.GET("/profiles", controllers.GetProfiles)
+		api.GET("/profiles/:id", controllers.GetProfile)
+		api.POST("/profiles", controllers.CreateProfile)
+		api.PUT("/profiles/:id", controllers.UpdateProfile)
+		api.DELETE("/profiles/:id", controllers.DeleteProfile)
+
```
Comment (awakia) [inline _example/router/router.go]: why is this moved? is the order random?

**Comment by dtan4** [comment]
I'm sorry but somehow CI has still failed... now inspecting.

**Comment by awakia** [comment]
I think the name "integration test" is misleading.

When I review this, I could not find any integration. Integration always need other side of component because you cannot integrate yourself.

My proposal is:

### Option 1.

Put this test into test. This is because this test is the most basic test of apig functionality. So, we want all apig developers to run this test.

### Option 2.

Rename this to generation-test or some others. I personally like option 1, but I understand the implementation difficulty of option 1. So just renaming solve the problem.

File: _example/router/router.go (hunk 1 above)
Comment (dtan4) [inline _example/router/router.go]: Now the order is random, because all models are fetched in parallel.

Random order causes extra diff of routers even if there is no diff of models. So models are sorted in ascii order beforehand.

**Comment by dtan4** [comment]
@awakia Removed all random generation logic and renamed to generation test. Please take a look.

**Comment by awakia** [comment]
Thanks, lgtm.
